### PR TITLE
Endpoints Tab | 05 | Style, Routes and Copy Command

### PR DESCRIPTION
**Pull request by bstavroulakis**
---
⚠️  _This branch depends on changes that will be introduced with #2422. For a list of changes please see: https://github.com/dcos/dcos-ui/compare/bill/feat/connection-endpoints/04-endpoint-tab-components...bill/feat/connection-endpoints/05-endpoint-route-styles_ 

---

Adds the routes necessary for the tab to appear in the pods and service details page.
Style changes for the endpoints tab.

**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by bstavroulakis** [comment]
#2422

File: plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js [hunk 1]
```diff
@@ -72,35 +101,50 @@ class ServiceConnectionContainer extends React.Component {
...
+            {this.getProtocolValue(portDefinition)}
+          </ConfigurationMapValue>
         </ConfigurationMapRow>
         <ConfigurationMapRow>
           <ConfigurationMapLabel>
             Container Port
           </ConfigurationMapLabel>
           <ConfigurationMapValue>
-            {getDisplayValue(portDefinition.containerPort)}
+            {!portDefinition.containerPort ||
```
Comment (nLight) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js]: please extract a function here

File: plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js [hunk 2]
```diff
@@ -72,35 +101,50 @@ class ServiceConnectionContainer extends React.Component {
...
+            {this.getHostPortValue(portDefinition)}
+          </ConfigurationMapValue>
         </ConfigurationMapRow>
         <ConfigurationMapRow>
           <ConfigurationMapLabel>
             Service Port
           </ConfigurationMapLabel>
           <ConfigurationMapValue>
-            {getDisplayValue(portDefinition.servicePort)}
+            {!portDefinition.servicePort || portDefinition.servicePort === ""
```
Comment (nLight) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js]: and here

File: plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js (hunk 1 above)
Comment (bstavroulakis) [reply plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js]: extracted

File: plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js (hunk 2 above)
Comment (bstavroulakis) [reply plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js]: extracted

**Comment by bstavroulakis** [comment]
Alfred test this

**Comment by bstavroulakis** [comment]
Alfred test this

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js [hunk 3]
```diff
@@ -11,59 +11,107 @@ import ConfigurationMapSection
...
+          </ClipboardTrigger>
+        </div>
+        {command}
+      </div>
+    );
+  }
+
   getProtocolValue(portDefinition) {
     let protocol = portDefinition.protocol || "";
+    if (protocol instanceof Array) {
```
Comment (weblancaster) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: why not `Array.isArray(protocol)`? I believe is more declarative.

File: src/styles/components/code/styles.less [hunk 4]
```diff
@@ -65,6 +75,11 @@
           right: @code-block-padding-right-screen-large;
           top: calc(@code-block-padding-top-screen-large ~' - 3px');
         }
+
+        .code-copy-icon.tight {
+          right: 0;
```
Comment (weblancaster) [inline src/styles/components/code/styles.less]: can't we re-use theses properties?

Comment (bstavroulakis) [reply src/styles/components/code/styles.less]: Re-use, meaning create a new generic class .right-0 or something for this?

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js [hunk 5]
```diff
@@ -11,59 +11,107 @@ import ConfigurationMapSection
...
 
-const METHODS_TO_BIND = ["handleOpenEditConfigurationModal"];
+const METHODS_TO_BIND = ["handleOpenEditConfigurationModal", "handleTextCopy"];
 
 class ServiceConnectionEndpointList extends React.Component {
   constructor() {
     super(...arguments);
 
+    this.state = {
+      copiedCommand: false
```
Comment (weblancaster) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: Should we change the value for `copiedCommand` here too?!

**Pull request merged**